src/application/queries/milo/get-sessions.jeune.milo.query.handler.db.ts

```diff
@@ -0,0 +1,137 @@
+import { StructureMiloSqlModel } from 'src/infrastructure/sequelize/models/structure-milo.sql-model'
+import { SessionJeuneDetailDto } from 'src/infrastructure/clients/dto/milo.dto'
+
+export interface GetSessionsJeuneMiloQuery extends Query {
+  idJeune: string
+  token: string
+}
+
+@Injectable()
+export class GetSessionsJeuneMiloQueryHandler extends QueryHandler<
+  GetSessionsJeuneMiloQuery,
+  Result<SessionJeuneMiloQueryModel[]>
+> {
+  constructor(
+    // TODO-178: Est-ce qu'on doit plutôt injecter le JeuneMiloRepo, et dupliquer la méthode get ??
+    @Inject(JeunesRepositoryToken)
+    private readonly jeuneRepository: Jeune.Repository,
+    private readonly keycloakClient: KeycloakClient,
+    private readonly miloClient: MiloClient,
+    private readonly jeuneAuthorizer: JeuneAuthorizer
+  ) {
+    super('GetSessionsJeuneMiloQueryHandler')
+  }
+
+  async handle(
+    query: GetSessionsJeuneMiloQuery
+  ): Promise<Result<SessionJeuneMiloQueryModel[]>> {
+    const jeune = await this.jeuneRepository.get(query.idJeune)
+    if (!jeune) {
+      return failure(new NonTrouveError('Jeune', query.idJeune))
+    }
+    if (!jeune.idPartenaire) {
+      return failure(new JeuneMiloSansIdPartenaire(query.idJeune))
+    }
+
+    const idpToken = await this.keycloakClient.exchangeTokenJeune(
+      query.token,
+      jeune.structure
+    )
+
+    const resultSessionMiloClient = await this.miloClient.getSessionsJeune(
+      idpToken,
+      jeune.idPartenaire
+    )
+
+    if (isFailure(resultSessionMiloClient)) {
+      return resultSessionMiloClient
+    }
+
+    return success(await sessionsVisible(resultSessionMiloClient.data.sessions))
+  }
+
+  async authorize(
+    query: GetSessionsJeuneMiloQuery,
+    utilisateur: Authentification.Utilisateur
+  ): Promise<Result> {
+    return this.jeuneAuthorizer.autoriserLeJeune(
+      query.idJeune,
+      utilisateur,
+      estMiloPassEmploi(utilisateur.structure)
+    )
+  }
+
+  async monitor(): Promise<void> {
+    return
+  }
+}
+
+async function sessionsVisible(
+  sessionsMiloClient: SessionJeuneDetailDto[]
+): Promise<SessionJeuneMiloQueryModel[]> {
+  return sessionsMiloDbVisible(sessionsMiloClient).then(sessionsDbVisible =>
+    sessionsMiloClient
+      .filter(session => estVisible(session, sessionsDbVisible))
+      .map(session => toQueryModel(session, sessionsDbVisible))
+  )
+}
+
+async function sessionsMiloDbVisible(
+  sessionsMiloClient: SessionJeuneDetailDto[]
+): Promise<SessionMiloSqlModel[]> {
+  return SessionMiloSqlModel.findAll({
+    where: {
+      id: sessionsMiloClient.map(session => session.session.id.toString()),
+      estVisible: true
+    },
+    include: [{ model: StructureMiloSqlModel, as: 'structure' }]
+  })
+}
+
+function estVisible(
+  session: SessionJeuneDetailDto,
+  sessionsDbVisible: SessionMiloSqlModel[]
+): boolean {
+  return sessionsDbVisible
+    .map(session => session.id)
```

Comment:
code smell, 'session' is already declared in the upper scope on line 114 column 3.